### PR TITLE
Addon-controls: Conditional arg disabling

### PR DESCRIPTION
**Pull request by shilman**
Issue: #11984

## What I did
Simplified disabling of individual args in ArgsTable, also disable controls conditionally on other controls:

```js
argTypes: {
  foo: { disable: true }, // disable
  bar: { table: { disable: true } }, // disable and log deprecation warning
  baz: { disable: 'foo' } // disable when foo is truthy
  moo: { disable: '!foo' } // disable when foo is falsey
}
```

Some examples:

conditional-controls

## How to test
See updated stories & unit tests

### CODE REVIEW

**Comment by tmeasday** [comment]
Does this have meaning outside of args table?

**Comment by shilman** [comment]
Not currently, but it could. Disabling args entirely could disable them from being serialized, or disable them from URL sync. WDYT @tmeasday @ghengeveld?

Rationale for this PR: `table.disable` is too verbose for something that is pretty common. I also merged another PR that makes removing args from the table more ergonomic #13898

**Comment by ghengeveld** [comment]
LGTM

File: docs/snippets/common/component-story-disable-controls.mdx.mdx [hunk 1]
```diff
@@ -8,10 +8,8 @@ import { YourComponent } from './your-component'
 <Meta 
   title='My Story' 
   argTypes={{
-    foo:{
-      table:{
-        disable:true
-      }
+    foo: {
+      disable:true
```
Comment (Vadorequest) [inline docs/snippets/common/component-story-disable-controls.mdx.mdx:12]: space

**Review by Vadorequest — APPROVED** [review]

**Review by ndelangen — APPROVED** [review]

**Comment by shilman** [comment]
Feedback from @ndelangen - let's make `disable: true` disable in the args data structure, not just in the table UI. Then this PR is self-consistent.

**Comment by shilman** [comment]
Question: transitive behavior?

**Comment by jorisw** [comment]
Why was this PR closed?

[This comment](https://github.com/storybookjs/storybook/issues/11984#issuecomment-780075168) best describes why this is needed.

**Comment by shilman** [comment]
@jorisw this PR was a POC. we haven't decided whether to add this feature to controls and plan to revisit the decision based on the community response.

**Pull request closed without merging**